docs/src/pages/ThemingSpecification.mdx

```diff
@@ -0,0 +1,301 @@
+export const meta = {
+  title: 'Theming specification',
+}
+
+## Goals and requirements
+
+The theming mechanism as part of Fluent UI should be flexible and provide support for CSSinJS theme, and all existing customization scenarios. It must also allow users to opt-out if the theming approach does not meet their needs. This means that we need to have support for both CSS stylesheets and CSSinJS.
+
+##### Support for CSS stylesheets
+
+Тhe users should be able to specify mappings from properties to classnames
+
+**IMPORTANT**: applications which load their own CSS in other means should not pay the penalty of loading a runtime CSS-in-JS package, especially if the class names are known at time of authoring.
+
+##### Support for CSSinJS
+
+CSSinJS support that will not tie the users to just one CSS in JS framework. The users should be able to define their styles and specify their engine that will be aligned to some defined interface.
```

Comment:
We are not going to support any compatibility layer for different engines, i.e. if user decide to use JSS instead of Fela that is shipped by default user should to make his styles compatible on his side.

It will be great to emphasize this.